servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieRest.java

```diff
@@ -634,4 +643,25 @@ public void paramConverterInvalidValue() {
     assertThat(nessieError.getMessage())
         .contains("No enum constant org.projectnessie.model.Reference.ReferenceType.X");
   }
+
+  @NessieApiVersions(versions = {NessieApiVersion.V2})
+  @ParameterizedTest
+  @CsvSource({
+    "-~1",
+    "main~1",
+    "main@cafebabe~1",
+    "-^2",
+    "main^2",
+    "main@cafebabe^2",
+    "-*2021-04-07T14:42:25.534748Z",
+    "main*2021-04-07T14:42:25.534748Z",
+    "main@cafebabe*2021-04-07T14:42:25.534748Z",
+  })
+  public void commitWithRelativeHashesNotAllowed(String ref) {
+    ContentKey key1 = ContentKey.of("test", "Key");
+    IcebergTable table1 = IcebergTable.of("loc1", 1, 2, 3, 4);
+    NessieError error =
+        prepareCommitV2(ref, key1, table1, 2, true).statusCode(400).extract().as(NessieError.class);
+    assertThat(error.getMessage()).startsWith("Relative hash not allowed here");
```

Comment:
from `curl` it looks like this:
```
$ curl -X 'POST' "http://localhost:19120/api/v2/trees/main@1f7ec3eb98eef851c5e4348144ec13ce658673fabd155aefbd862be5cf07f88e~1/history/commit" [...]

{
  "status" : 400,
  "reason" : "Bad Request",
  "message" : "Relative hash not allowed here: 1f7ec3eb98eef851c5e4348144ec13ce658673fabd155aefbd862be5cf07f88e~1",
  "errorCode" : "BAD_REQUEST",
  "serverStackTrace" : null
}
```
This is much better than before, but still "here" is a bit obscure, IMHO... Could we say something like `Relative hash not allowed in commit operations: 1f7ec3eb98eef851c5e4348144ec13ce658673fabd155aefbd862be5cf07f88e~1`?